public/scripts/backgrounds.js

```diff
@@ -81,15 +82,20 @@ async function forceSetBackground(backgroundInfo) {
 }
 
 async function onChatChanged() {
-    if (hasCustomBackground()) {
-        setCustomBackground();
-    }
-    else {
-        unsetCustomBackground();
+    const lockedUrl = chat_metadata[BG_METADATA_KEY];
+
+    if (lockedUrl) {
+        // This chat has a locked background, so apply it directly to the main view.
+        $('#bg1').css('background-image', lockedUrl);
```

Comment:
Could get rid of the outer if

```suggestion
        $('#bg1').css('background-image', lockedUrl || background_settings.url);
```